go/datas/database.go

```diff
@@ -15,8 +15,7 @@ import (
 // Database provides versioned storage for noms values. Each Database instance represents one moment in history. Heads() returns the Commit from each active fork at that moment. The Commit() method returns a new Database, representing a new moment in history.
 type Database interface {
 	// To implement types.ValueWriter, Database implementations provide WriteValue(). WriteValue() writes v to this Database, though v is not guaranteed to be be persistent until after a subsequent Commit(). The return value is the Ref of v.
-	types.ValueWriter
-	types.ValueReader
+	types.ValueReadWriter
```

Comment:
This means that anything on `ValueReadWriter` becomes a part of the Database interface. This includes unexported methods, but they will only be visible in the package where `ValueReadWriter` is defined. Thus, `ValueReadWriter` can declare something unexported (e.g. `opCache() *opCache`), `ValueStore` can implement it, and while it will be a (masked) part of the Database API, the only code that can use it is stuff inside `types` that takes a ValueReadWriter, such as `Map.NewMutator()`.